addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -52,7 +52,8 @@ var LinkDialog = Dialog.extend({
 
         if (this.data.range) {
             this.data.iniClassName = $(this.data.range.sc).filter("a").attr("class") || "";
-            this.data.className = this.data.iniClassName.replace(/(^|\s+)btn(-[a-z0-9_-]*)?/gi, ' ');
+            this.data.className = this.data.iniClassName.replace(/(^|\s+)btn(-[a-z0-9_-]*)?/gi, ' ')
+                .replace(/\bflat\b/gi, '').replace(/\brounded-circle\b/gi, '');;
```

Comment:
> Also avoid accumulating non-"btn*" classes (flat & rounded-circle).

I cannot reproduce the issue, are you sure this is necessary? And if it is, maybe do the replace in one regex?

Reply:
It does not happen in 13.0 indeed. I had the issue in 12.0 before re-targeting and I forgot to verify if that part of the problem still existed.